### PR TITLE
Feature/model creator

### LINKED ISSUE #283: Model Creator: Modifers not created
A `modifiers.madx` file is assumed, when loading model from directory, 
but this file is never written out by the creator. 
Instead the path to the modifiers is inserted into the `job.madx`.

Also there is no problem allowing multiple modifier files. 

possible solutions: 
 - read modifier files from `job.madx` by a common tag
 - create `modifiers.madx` file and read modifiers from there.

### LINKED ISSUE #284: Model Creator: Allow only nat-tunes given if no excitation is selected
This will drop the 
```
if (acd == 0) ...
```
parts of the `job.madx`, which makes it less flexible.

### LINKED ISSUE #285: Model Creator: Create and use accel.get_base_madx_script
The `get_base_madx_script` function is mandatory for all accelerators and will be used by the fullresponse creation.
So far it is only implemented for `lhc`.

### PR DESCRIPTION
**Pull request by JoschD**
Fixes #283 
Fixes #284 
Fixes #285 
Changes towards #238 for model creation, accelerator class and where they were used (e.g. kmod-analysis, madx_wrapper)

Hint: In kmod I also changed the `--beam` input from `"B1"` or `"B2"` to `1` or `2`, to be aligned with `--beam` in accelerator class

### CODE REVIEW

File: omc3/model/accelerators/accelerator.py [hunk 1]
```diff
@@ -285,3 +320,23 @@ class AcceleratorDefinitionError(Exception):
...
+
+
+# Helper ----
+
+def _get_modifiers_from_modeldir(model_dir: Path):
+    """Parse modifiers from job.create_model.madx or use modifiers.madx file."""
+    job_file = model_dir / JOB_MODEL_MADX
+    if job_file.exists():
+        job_madx = job_file.read_text()
+        modifiers = re.findall(fr"\s+call,\s*file\s*=\s*[\"\']?([^;\'\"]+)[\"\']?\s*;\s*{MODIFIER_TAG}", job_madx,
```
Comment (fsoubelet) [inline omc3/model/accelerators/accelerator.py:332]: Could use a comment to explain that regex, so we don't need to think about it too much later on

File: omc3/model/accelerators/lhc.py [hunk 2]
```diff
@@ -240,19 +240,19 @@ def get_ips(self):
                    Lhc.DOROS_IP_BPMS.format(side="R", ip=ip, beam=self.beam))
 
     def log_status(self):
-        LOGGER.info(f"  model dir = {self.model_dir}")
-        LOGGER.info("Natural Tune X      [{:10.3f}]".format(self.nat_tunes[0]))
-        LOGGER.info("Natural Tune Y      [{:10.3f}]".format(self.nat_tunes[1]))
-        LOGGER.info("Best Knowledge Model     [{:>10s}]".format(
+        LOG.info(f"  model dir = {self.model_dir}")
+        LOG.info("Natural Tune X      [{:10.3f}]".format(self.nat_tunes[0]))
```
Comment (fsoubelet) [inline omc3/model/accelerators/lhc.py:244]: let's use f-strings everywhere here

File: omc3/model/accelerators/ps/base.mask [hunk 3]
```diff
@@ -79,42 +79,3 @@ match, use_macro;
...
 ENDMATCH;
-
-
-
-/******************************************************************************************
- * TWISS
- ******************************************************************************************/
-
-
-/************************************************************
```
Comment (fsoubelet) [inline omc3/model/accelerators/ps/base.mask:90]: Why are we getting rid of this block?

Comment (fsoubelet) [reply omc3/model/accelerators/ps/base.mask:90]: Nevermind I see it's moved in a different file.

File: omc3/model_creator.py [hunk 4]
```diff
@@ -20,8 +20,10 @@
 from omc3.model.model_creators.psbooster_model_creator import PsboosterModelCreator
 from omc3.model.model_creators.segment_creator import SegmentCreator
 from omc3.utils.iotools import create_dirs
+from omc3. utils import logging_tools
```
Comment (fsoubelet) [inline omc3/model_creator.py:23]: typo space in import

File: omc3/model_creator.py [hunk 5]
```diff
@@ -37,15 +39,23 @@
...
-                         help="Output path for model, twiss files will be writen here.")
-    params.add_parameter(name="writeto", type=str,
-                         help="Path to the file where to write the resulting MAD-X script.")
-    params.add_parameter(name="logfile", type=str,
-                         help=("Path to the file where to write the MAD-X script output."
-                               "If not provided it will be written to sys.stdout."))
+    params.add_parameter(
+        name="type",
+        choices=("nominal", "best_knowledge", "coupling_correction"),
+        help="Type of model to create, either nominal or best_knowledge"
```
Comment (fsoubelet) [inline omc3/model_creator.py:45]: Help does not reflect choices

File: omc3/model_creator.py [hunk 6]
```diff
@@ -110,26 +115,18 @@ def create_instance_and_model(opt, accel_opt):
 
         JPARC: Not implemented
     """
-    if sys.flags.debug:
```
Comment (fsoubelet) [inline omc3/model_creator.py:113]: Removing the option to enable debug logging? :(

Comment (fsoubelet) [reply omc3/model_creator.py:113]: Nevermind again, I see the logic has been moved to logging_tools 👍

File: tests/unit/test_model_creator.py [hunk 7]
```diff
@@ -1,86 +1,155 @@
...
-                    writeto=join(BASE_OUTPUT, "job.twiss_best_knowledge.madx"),
-                    logfile=join(BASE_OUTPUT, "madx_log_best_knowledge.txt"))
-    create_instance_and_model(opt_dict)
-    _clean_up(BASE_OUTPUT)
-
-
-def _clean_up(path_dir):
-    if isdir(path_dir):
-        rmtree(path_dir, ignore_errors=True)
+def test_lhc_creation_best_knowledge(tmp_path):
```
Comment (fsoubelet) [inline tests/unit/test_model_creator.py:103]: This test does not test the creation.

**Review by fsoubelet — CHANGES_REQUESTED** [review]
Just a small omission as I can tell, otherwise LGTM

File: tests/unit/test_model_creator.py (hunk 7 above)
Comment (JoschD) [reply tests/unit/test_model_creator.py:103]: Ooops. It did at some point

**Review by tpersson — APPROVED** [review]
I had a look and no comments on the changes but just observations that still are a lot of things missing such as a real creation of the segment-by-segment but that is fine and I will have a look at that shortly. I think it is crucial that this PR is merged so we can merge the correction branch.

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**